R/plotCCA.R

```diff
@@ -0,0 +1,504 @@
+#' Plot RDA or CCA object
+#'
+#'
+#' @param object a
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}, a
+#'   \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-constructor]{TreeSummarizedExperiment}}
+#'   or a rda/cca object. The latter is the output of \code{\link[mia:runCCA]{calculateRDA}}.
+#' 
+#' @param dimred A string or integer scalar indicating the reduced dimension to
+#'   plot. This is the output of \code{\link[mia:runCCA]{runRDA}} and resides in
+#'   \code{reducedDim(tse, dimred)}.
+#'
+#' @param colour_by String specifying the name of a column in colData to colour by
+#'   (Default: NULL).
+#' 
+#' @param color_by Alias for `colour_by`.
+#' 
+#' @param ellipse_fill if TRUE, ellipses are filled. (Default: TRUE).
+#'
+#' @param alpha Number between 0 and 1 to adjust the opacity of ellipses (Default: 0.2).
+#'
+#' @param ellipse_size Number specifying the size of ellipses (Default: 0.1).
+#' 
+#' @param ellipse_linetype Discrete number specifying the style of ellipses (Default: 1).
+#' 
+#' @param vec_size Number specifying the size of vectors (Default: 0.5).
+#' 
+#' @param vec_colour String specifying the colour of vectors (Default: "black").
+#' 
+#' @param vec_color Alias for `vec_colour`.
+#' 
+#' @param vec_linetype Discrete number specifying the style of vector lines (Default: 1).
+#' 
+#' @param arrow_size Number specifying the size of arrows (Defaults: 0.25).
+#' 
+#' @param text_size Number specifying the size of text and labels (Default: 4).
+#' 
+#' @param text_colour String specifying the colour of text and labels (Default: "black").
+#' 
+#' @param text_color Alias for `text_colour`.
+#' 
+#' @param vec_text If TRUE, text instead of labels are used to label vectors (Default: TRUE).
+#' 
+#' @param repel_text If TRUE, labels are repelled (Default: TRUE).
+#' 
+#' @param ... additional parameters for plotting, inherited from
+#'   \code{\link[scater:plotReducedDim]{plotReducedDim}},
+#'   \code{\link[ggplot2:geom_label]{geom_label}} and
+#'   \code{\link[ggrepel:geom_label_repel]{geom_label_repel}}.
+#' 
+#' @details
+#' plotRDA and plotCCA create an RDA/CCA plot starting from the output of
+#' \code{\link[mia:runCCA]{CCA and RDA}} functions, two common methods for
+#' supervised ordination of microbiome data. Both
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' and rda/cca objects are supported as input. When the input is a SummarizedExperiment
```

Comment:
The input must be SingleCellExperiment (SE does not have reducedDim slot)

Also rda object is not supported. Matrix is supported (which is an output of calculateRDA)

Reply:
rda object can be found from the matrix attributes though...